metadata_backend/api/middlewares.py

```diff
@@ -147,9 +146,10 @@ async def create_session_with_token(req: web.Request) -> aiohttp_session.Session
     else:
         return session  # current empty session
 
-    # Get path to userinfo from OIDC config
-    aai = ServiceHandler(base_url=URL(aai_config["oidc_url"].rstrip("/")))
+    # we need to create a new instance every time, this is not something we can re-use
+    aai = AAIServiceHandler()
```

Comment:
@teemukataja if you have different opinions on how to manage this, we can change.

Reply:
We discussed that this is probably not possible